CMakeLists.txt

```
@@ -5,6 +5,16 @@ cmake_minimum_required(VERSION 3.22)
 set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
 project(msvc_standard_libraries LANGUAGES CXX)
 
+include(CheckCXXSourceCompiles)
+check_cxx_source_compiles([=[
+#include <sdkddkver.h>
+static_assert(WDK_NTDDI_VERSION >= NTDDI_WIN10_CO, "Inspecting WDK_NTDDI_VERSION, the Windows SDK version.");
+int main() {}
+]=] WINDOWS_SDK_VERSION_CHECK)
+if(NOT WINDOWS_SDK_VERSION_CHECK)
+    message(FATAL_ERROR "To build the STL, select the Windows 11 SDK (10.0.22000.0) in the VS Installer.")

```

nit:

```suggestion
    message(FATAL_ERROR "The STL must be built with the Windows 11 SDK or later. Make sure it's available by installing the Windows 11 SDK (at least version 10.0.22000.0) in the VS Installer.")
```

I think this error message is more obvious.

---

Hmm, I agree that the additional explanation is an improvement. However, I'd like to rephrase it to be future-proof - currently we require the first version of the Windows 11 SDK, but there will be more, and the message should remain accurate if we just increase the number. Will push a change...